src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let newString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        if (i !== 0) {
+          newString += separator;
+        }
+        continue;
+      }
+
+      if (i !== 0) {
+        newString += separator;
+      }
+      newString += this[i];
```

Comment:
I think it would be much easier to check if this[i] is correct and then perform operation instead of handling all different cases:
```suggestion
      if (this[i] !== null || this[i] !== undefined) {
          // add letter to newString
      }
      
      if(iterator is smaller than word length) // add separator
```